drivers/sensor/xbr818/xbr818.c

```diff
@@ -0,0 +1,434 @@
+	__ASSERT(ret == 0, "Error sending XBR818 defaults for XBR818_RF_EN_SEL");
+	ret |= i2c_reg_write_byte_dt(&config->i2c, XBR818_SAMPLE_RATE_DIVIDER, 0x20);
+	__ASSERT(ret == 0, "Error sending XBR818 defaults for XBR818_SAMPLE_RATE_DIVIDER");
+	ret |= i2c_reg_write_byte_dt(&config->i2c, XBR818_RF_POWER, 0x45);
+	__ASSERT(ret == 0, "Error sending XBR818 defaults for XBR818_RF_POWER");
+	ret |= i2c_reg_write_byte_dt(&config->i2c, XBR818_TIMER_CTRL, 0x21);
+	__ASSERT(ret == 0, "Error sending XBR818 defaults for XBR818_TIMER_CTRL");
+
+	data[0] = 0x5a;
+	data[1] = 0x01;
+	ret |= i2c_burst_write_dt(&config->i2c, XBR818_THRESHOLD_1, data, 2);
+	__ASSERT(ret == 0, "Error sending XBR818 defaults for XBR818_THRESHOLD");
+
+	data[0] = 0x55;
+	data[1] = 0x01;
+	ret |= i2c_burst_write_dt(&config->i2c, XBR818_THRESHOLD_NOISE_1, data, 2);
+	__ASSERT(ret == 0, "Error sending XBR818 defaults for XBR818_THRESHOLD_NOISE");
+
+	/* 0.1 seconds */
+	data[0] = 0x80;
+	data[1] = 0x0C;
+	data[2] = 0x00;
+	ret |= i2c_burst_write_dt(&config->i2c, XBR818_DELAY_TIME_1, data, 3);
+	__ASSERT(ret == 0, "Error sending XBR818 defaults for XBR818_DELAY_TIME");
+
+	/* 0.5 seconds */
+	data[0] = 0x80;
+	data[1] = 0x3E;
+	data[2] = 0x00;
+	ret |= i2c_burst_write_dt(&config->i2c, XBR818_LOCK_TIME_1, data, 3);
+	__ASSERT(ret == 0, "Error sending XBR818 defaults for XBR818_LOCK_TIME");
+
+	ret |= i2c_reg_write_byte_dt(&config->i2c, XBR818_PIN_SETTINGS, 0x0C);
+	__ASSERT(ret == 0, "Error sending XBR818 defaults for XBR818_PIN_SETTINGS");
+
+	ret |= i2c_reg_write_byte_dt(&config->i2c, XBR818_I2C_OUT, 0x1);
+	__ASSERT(ret == 0, "Error sending XBR818 defaults for XBR818_I2C_OUT");
+
+	return ret;
+}
+
+static int xbr818_init(const struct device *dev)
+{
+	const struct xbr818_config *config = dev->config;
+	struct xbr818_data *data = dev->data;
+	int ret;
+
+	if (!i2c_is_ready_dt(&config->i2c)) {
+		LOG_ERR("I2C device not ready");
+		return -ENODEV;
+	}
+
+	data->dev = dev;
+	data->work.handler = xbr818_work;
+
+	ret = gpio_pin_configure_dt(&config->io_val, GPIO_INPUT);
+	if (ret != 0) {
+		LOG_ERR("%s: could not configure io_val(int) pin", dev->name);
+		return ret;
+	}
+
+	if (config->i2c_en.port) {
+		ret = gpio_pin_configure_dt(&config->i2c_en, GPIO_OUTPUT);
+		if (ret != 0) {
+			LOG_ERR("%s: could not configure i2c_en pin", dev->name);
+			return ret;
+		}
+	}
+
+	ret = xbr818_enable_i2c(dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = xbr818_init_defaults(dev);
+	if (ret != 0) {
+		LOG_ERR("%s: unable to configure", dev->name);
+	}
+
+	ret = xbr818_disable_i2c(dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = gpio_pin_interrupt_configure_dt(&config->io_val, GPIO_INT_DISABLE);
+	if (ret) {
+		LOG_ERR("%s: failed to configure gpio interrupt: %d", dev->name, ret);
+		return ret;
+	}
+
+	gpio_init_callback(&data->gpio_cb, xbr818_gpio_callback, BIT(config->io_val.pin));
+
+	return ret;
+}
+
+static const struct sensor_driver_api xbr818_api = {
```

Comment:
```suggestion
static DEVICE_API(sensor, xbr818_api) = {
```

Reply:
Should I submit a fix commit for this given this was merged before i had the time to update it?

Reply:
It is at https://github.com/zephyrproject-rtos/zephyr/pull/85762